Fix shuttles

Chryseus opened the pull request:
Fixes shuttle creation and flight.

Commit:
Aghost can now change job assignment

Commit:
Fix map loading

Commit:
Merge https://github.com/Persistent-SS13/Persistent-Bay into dev

Commit:
Fix pizza boxes so they don't respawn

Andrew-Fall committed:
Merge branch 'dev' into dev

Commit:
Merge Persistent-SS13/Persistent-Bay/dev into Chryseus/Persistent-Bay/dev

Commit:
Shuttles now mostly working

Commit:
More work on shuttle

Commit:
More work

Commit:
Finally working

Commit:
Merge branch 'dev' into shuttlefix

Commit:
Cleanup

Commit:
Merge

michaelchessall committed:
Merge branch 'dev' into shuttlefix

michaelchessall commented on code/modules/shuttles/shuttle.dm:
```diff
@@ -176,42 +177,62 @@
 				var/turf/TA = GetAbove(TO)
-				if(istype(TA, ceiling_type))
-					TA.ChangeTurf(get_base_turf_by_area(TA), 1, 1)
+				if(istype(TA, ceiling_type) || istype(TA, /turf/simulated/floor/airless) || istype(TA, /turf/simulated/open))
```
This section is problematic as heck. I'm worried about shuttles landing underneath constructed areas and tearing them apart when they leave. This also deletes the contents of the shuttles roof, which is not a good idea, in any circumstance.

Chryseus replied:
You're right, I'll update it so it works properly in all situations.

Chryseus replied:
I'm quite happy with how the shuttles work now, docks have a new variable for interior or exterior which should be saved, items left on the roof are now also kept, although it might make more sense for them to be knocked away when the shuttle moves.

Andrew-Fall committed:
Merge branch 'dev' into shuttlefix

Andrew-Fall committed:
Merge branch 'dev' into shuttlefix

michaelchessall committed:
Merge branch 'dev' into shuttlefix

Commit:
Merge remote-tracking branch 'origin/dev' into shuttlefix

# Conflicts:
#	code/modules/shuttles/shuttle.dm

Commit:
Preserve contents of shuttle ceiling.

Chryseus committed:
Merge branch 'dev' into shuttlefix

Chryseus committed:
Merge branch 'dev' into shuttlefix

michaelchessall committed:
Merge branch 'dev' into shuttlefix

Chryseus committed:
Merge branch 'dev' into shuttlefix

Commit:
Merge remote-tracking branch 'origin/dev' into shuttlefix

Commit:
Final commit probably

Chryseus committed:
Merge branch 'dev' into shuttlefix

Commit:
Revert "Final commit probably"

This reverts commit 1624c1f37ea82557bb8053b1e7f4cf03d92c0a33.

Commit:
Fixing branch

Commit:
Merge remote-tracking branch 'origin/dev' into shuttlefix

Commit:
Merge remote-tracking branch 'chryseus/shuttlefix' into shuttlefix

Commit:
Remove garbage

Chryseus committed:
Merge branch 'dev' into shuttlefix

Commit:
Revert "Final commit probably"

This reverts commit 1624c1f37ea82557bb8053b1e7f4cf03d92c0a33.

Commit:
Merge 'origin/dev'

Chryseus committed:
Merge branch 'dev' into shuttlefix

Commit:
Docking beacon resets when unwrenched.

Commit:
Merge

Chryseus commented:
**Current major issues remaining**

- External docks need to check for non-space areas above to avoid causing damage.
- Have docking beacons reset properly when moved

**Features being worked on for a later PR**
- Allow arbitrary shuttle sizes (within reason)
- Rotation of shuttles
- Fuel consumption, internal / external fuel port, etc

Chryseus committed:
Merge branch 'dev' into shuttlefix

Commit:
Merge remote-tracking branch 'chryseus/dev' into shuttle_bugfixing

Commit:
Shuttle no longer damages things above it under any condition
Code cleaned up and comments added
Removed debugging messages
ChangeTurf has an additional parameter, var/space_override = 0 to override /turf/space behavior
Interior lighting update now applies when arriving at an interior dock

Commit:
Merge remote-tracking branch 'chryseus/shuttle_bugfixing' into shuttlefix

Chryseus commented:
PR is ready for review again.
Extensive testing has been done and I can't get it to break so that's a good sign at least.
